src/stream/src/executor/merge.rs

```diff
@@ -109,16 +110,25 @@ impl MergeExecutor {
                 .actor_input_buffer_blocking_duration_ns
                 .with_label_values(&[&actor_id_str, &upstream_fragment_id_str])
                 .inc_by(start_time.elapsed().as_nanos() as u64);
-            let msg: Message = msg?;
+            let mut msg: Message = msg?;
 
-            match &msg {
+            match &mut msg {
                 Message::Chunk(chunk) => {
                     self.metrics
                         .actor_in_record_cnt
                         .with_label_values(&[&actor_id_str])
                         .inc_by(chunk.cardinality() as _);
                 }
                 Message::Barrier(barrier) => {
+                    if cfg!(debug_assertions) {
```

Comment:
`trace` will be disabled in release mode, so we don't need to wrap `if cfg!(debug_assertions)`. Also please set the target to `events::barrier::path` (instead of `risingwave_stream::executor::merge`) so that we can easily filter them out.